### PR TITLE
[WIP] add data retention capability

### PR DESCRIPTION
**Pull request by tomkralidis**

### CODE REVIEW

**Comment by petersilva** [comment]
hate that... in public the organization of the directories is by date for a reason.
having a find traverse the entire tree is expensive.  Just look at the root directories and unconditionally delete the tree under there.  Many fewer IO operations.

I would think the incoming would be emptied out by the csv2bufr deleting the input after the output is created?

Not sure how OUTGOING is managed.

**Comment by tomkralidis** [comment]
For PUBLIC, sure.  What about beyond PUBLIC?  We can't assume that INCOMING would have a date-based directory structure?  @david-i-berry @chenxiaoxia2019 

OUTGOING is currently unused (we were thinking to use it for API publishing, but this has gone another way).  We may remove OUTGOING as a thing soon.

**Comment by petersilva** [comment]
I guess it's fine for now... can always be improved later.

**Review by petersilva — APPROVED** [review]
OK for now... would prefer a more efficient approach to PUBLIC, 
and I think old files being in INCOMING is a sign of a problem/error... but for dev puropses this is fine for now.

**Review by webb-ben — APPROVED** [review]

File: docker/wis2node/wis2node.cron [hunk 1]
```diff
@@ -0,0 +1,9 @@
+### Clean out files 
+0 * * * * /usr/bin/find $WIS2NODE_DATADIR_INCOMING -type f -mtime +$WIS2NODE_DATADIR_RETENTION_DAYS -exec rm {} \; > /dev/null 2>&1
+0 * * * * /usr/bin/find $WIS2NODE_DATADIR_OUTGOING -type f -mtime +$WIS2NODE_DATADIR_RETENTION_DAYS -exec rm {} \; > /dev/null 2>&1
+0 * * * * /usr/bin/find $WIS2NODE_DATADIR_PUBLIC -type f -mtime +$WIS2NODE_DATADIR_RETENTION_DAYS -exec rm {} \; > /dev/null 2>&1
+
+### Clean out empty directories
+0 * * * * /usr/bin/find $WIS2NODE_DATADIR_INCOMING -type d -empty -delete > /dev/null 2>&1
```
Comment (david-i-berry) [inline docker/wis2node/wis2node.cron:7]: I'm  not sure we want to delete the incoming data at this stage, especially once we start pointing at the s3 bucket / ftp server. I would suggest adding an archive directory where the incoming data can be moved to once successfully processed.

Comment (petersilva) [reply docker/wis2node/wis2node.cron:7]: @david-i-berry that's fine... I just don't want an input directory to grow without bound... the most efficient input directory is one that is kept empty.   Putting the data somewhere else is fine.  could even just copy the input into public, and avoid creating another directory.

Comment (david-i-berry) [reply docker/wis2node/wis2node.cron:7]: Moving to an archive directory, zipping and taring would be my choice in the short term. In the longer term the appropriate archive for the incoming data needs to be identified.

Comment (tomkralidis) [reply docker/wis2node/wis2node.cron]: For now, I've commented out any removals from incoming.  For longer term discussion.

**Comment by tomkralidis** [comment]
Closing in lieu of #75.

**Pull request closed without merging**